### ISSUE TITLE
getDecimalsForValue function stripping out decimals from negative decimal numbers

### ISSUE THREAD

**Issue by yosiasz** [grafana/grafana#53331]
**What happened**:
negative decimal numbers' decimals get dropped

**What you expected to happen**:
i expect the auto decimal setting to respect my numbers :)

**How to reproduce it (as minimally and precisely as possible)**:

using microsoft sql server as data source and fake data
```
select getdate() time, -27.5 value
```
The value is rounded to -27 with decimal set to auto. if one changes it to 1 then it works. but auto should respect my numbers.

Self debugging getDecimalsForValue in dev console.

![image](https://user-images.githubusercontent.com/900951/182951049-d08ab46d-7547-40ed-8855-957ece665b0e.png)

The issue is here

https://github.com/grafana/grafana/blob/2107652c3fdeebf499c360795c219f68ef47b65f/packages/grafana-data/src/valueFormats/valueFormats.ts#L82

getDecimalsForValue function. Also looks like it is missing a unit test for neg values.

**Anything else we need to know?**:

**Environment**:
- Grafana version: 9.0.3
- Data source type & version: inline csv
- OS Grafana is installed on: Windows
- User OS & Browser: Windows/Chrome
- Grafana plugins: infinity
- Others:

**Comment by joshhunt** [comment]
Reproduction dashboard attached.

![image](https://user-images.githubusercontent.com/46142/183403766-f934d171-1afd-4b58-b442-1ff17219f648.png)

Manually setting "Standard options > Decimals" to a number of decimal places to include doesn't chop of the decimal places (as expected), but auto should leave them in tact for negative numbers.


<details>
<summary>Dashboard JSON</summary>

```json
{
  "annotations": {
    "list": [
      {
        "builtIn": 1,
        "datasource": {
          "type": "grafana",
          "uid": "-- Grafana --"
        },
        "enable": true,
        "hide": true,
        "iconColor": "rgba(0, 211, 255, 1)",
        "name": "Annotations & Alerts",
        "target": {
          "limit": 100,
          "matchAny": false,
          "tags": [],
          "type": "dashboard"
        },
        "type": "dashboard"
      }
    ]
  },
  "editable": true,
  "fiscalYearStartMonth": 0,
  "graphTooltip": 0,
  "id": 619,
  "links": [],
  "liveNow": false,
  "panels": [
    {
      "datasource": {
        "type": "testdata",
        "uid": "ZQvVNvaMk"
      },
      "fieldConfig": {
        "defaults": {
          "color": {
            "mode": "thresholds"
          },
          "custom": {
            "align": "auto",
            "displayMode": "auto",
            "inspect": false
          },
          "mappings": [],
          "thresholds": {
            "mode": "absolute",
            "steps": [
              {
                "color": "green",
                "value": null
              },
              {
                "color": "red",
                "value": 80
              }
            ]
          }
        },
        "overrides": []
      },
      "gridPos": {
        "h": 9,
        "w": 12,
        "x": 0,
        "y": 0
      },
      "id": 2,
      "options": {
        "footer": {
          "fields": "",
          "reducer": [
            "sum"
          ],
          "show": false
        },
        "showHeader": true
      },
      "pluginVersion": "9.2.0-pre",
      "targets": [
        {
          "datasource": {
            "type": "testdata",
            "uid": "ZQvVNvaMk"
          },
          "refId": "A",
          "scenarioId": "csv_metric_values",
          "stringInput": "-27.5,27.5"
        }
      ],
      "title": "Panel Title",
      "type": "table"
    }
  ],
  "schemaVersion": 37,
  "style": "dark",
  "tags": [],
  "templating": {
    "list": []
  },
  "time": {
    "from": "now-6h",
    "to": "now"
  },
  "timepicker": {},
  "timezone": "",
  "title": "#53331 stripping out decimals from negative decimal numbers",
  "uid": "6hKZdki4k",
  "version": 2,
  "weekStart": ""
}
```

</details>

**Comment by JoaoSilvaGrafana** [comment]
Hi @yosiasz, thanks for raising this!
I want to understand how exactly you would expect it to work, we can make the negative numbers work just like the positive ones, however, the `auto` option still tries to be clever in calculating the decimals, for instance `27.52` gets converted to `27.5`, so even with a fix for the negatives, if you had `-27.52` it would only display `-27.5` and not `-27.52`.
For your issue, did you want the negative numbers to display just like the positives when `auto` is selected OR did you just want no decimals to be dropped at all?

Basically option A or option B in this column:

| value | current | option A | option B |
| --- | --- | --- | --- |
| 27.52 | 27.5 | 27.5 | 27.52 |
| -27.52 | -28 | -27.5 | -27.52 |

**Comment by yosiasz** [comment]
Hello @JoaoSilvaGrafana did you take a look at the code snipet Ibposted where the possible error exists?

Is there a bug there?

**Comment by JoaoSilvaGrafana** [comment]
> Hello @JoaoSilvaGrafana did you take a look at the code snipet I posted where the possible bug exists?
> 
> Is there a bug there? I don't think ot really matters what I want 😄
> 
> The code should work as it should. The same number in the positive 27.5 passing through that functiom stays intact. Why is a negative number -27.5 truncated to -27

Yes I agree with this and we can definitely fix that bug. I was just wondering if you wanted more precision than that, because that function basically will keep the number down to 3 digits when possible, like this:

<img width="480" alt="Screenshot 2022-08-19 at 14 30 33" src="https://user-images.githubusercontent.com/100691367/185629658-5e74b76f-ccba-4444-8797-80cea625cd6f.png">

If you wanted to _always_ keep your decimals then we would also have a workaround for that, wanted to know if that would be helpful to you or not 😄

**Comment by JoaoSilvaGrafana** [comment]
I have fixed the original bug that you raised @yosiasz 🙌 
I don't think there is any documentation about how auto works internally, but josh has another PR that he mentioned here to add a description to that Decimals option.
